src/app/feature/consultant/services/consultant-store/consultant-store.service.ts

```diff
@@ -69,13 +69,58 @@ export class ConsultantStore implements OnDestroy {
   /**
    * Build a new consultant object from existing object with updated properties.
    * @param data A partial Consultant object containing the data to update
+   * @param index optional arg which says which index in edu or cert array to replace
    */
-  private updatedConsultantFactory(data: Partial<Consultant>): Consultant {
+  private updatedConsultantFactory(data, index?: number): Consultant {
```

Comment:
data no longer matches the object Consultant because I added some extra properties. that is one of the check boxes above that I want to fix. Ideally I pass in a Consultant object so hopefully I can keep track of eduOrCert somewhere else.